### ISSUE TITLE
Querying deb_packages/apt_sources tables randomly causes extensions_socket file missing

### ISSUE THREAD

**Issue by briefash** [osquery/osquery#7602]
# Bug report

### What operating system and version are you using?

```
root@u1:/userap# osqueryi --line "SELECT version, build, platform FROM os_version;"
 version = 22.04 LTS (Jammy Jellyfish)
   build =
platform = ubuntu
root@u1:/userap#
```

### What version of osquery are you using?

```
root@u1:/userap# osqueryi --line "SELECT version from osquery_info;"
version = 5.2.3
root@u1:/userap#
```

### What steps did you take to reproduce the issue?

1. check extensions_socket file path of current osqueryi session, which is also used for the following steps
```
osquery> select value from osquery_flags where name = 'extensions_socket';
+-------------------------+
| value                   |
+-------------------------+
| /root/.osquery/shell.em |
+-------------------------+
osquery>
```
2. check extensions_socket file exists
```
osquery> select path from file where path = '/root/.osquery/shell.em';
+-------------------------+
| path                    |
+-------------------------+
| /root/.osquery/shell.em |
+-------------------------+
osquery>
```
3. query deb_packages table - 1
```
osquery> select count(1) from deb_packages;
+----------+
| count(1) |
+----------+
| 107      |
+----------+
osquery>
```
4. extensions_socket file exists
```
osquery> select path from file where path = '/root/.osquery/shell.em';
+-------------------------+
| path                    |
+-------------------------+
| /root/.osquery/shell.em |
+-------------------------+
osquery>
```
5. query deb_packages table - 2
```
osquery> select count(1) from deb_packages;
+----------+
| count(1) |
+----------+
| 107      |
+----------+
osquery>
```
6. extensions_socket file exists
```
osquery> select path from file where path = '/root/.osquery/shell.em';
+-------------------------+
| path                    |
+-------------------------+
| /root/.osquery/shell.em |
+-------------------------+
osquery>
```
7. query deb_packages table - 3
```
osquery> select count(1) from deb_packages;
+----------+
| count(1) |
+----------+
| 107      |
+----------+
osquery>
```
8. extensions_socket file exists
```
osquery> select path from file where path = '/root/.osquery/shell.em';
+-------------------------+
| path                    |
+-------------------------+
| /root/.osquery/shell.em |
+-------------------------+
osquery>
```
9. query deb_packages table - 4
```
osquery> select count(1) from deb_packages;
+----------+
| count(1) |
+----------+
| 107      |
+----------+
osquery>
```
10. extensions_socket file is missing
```
osquery> select path from file where path = '/root/.osquery/shell.em';
osquery>
```

### What did you expect to see?

The extensions_socket file should exist until the osqueryi session quits.

### What did you see instead?

The extensions_socket file is missing after querying deb_packages (apt_sources also) for random times.

### More information
(1) This issue can be reproduced in at least 5.2.3, 5.2.2, and 5.1.0. However in 5.1.0, the extensions_socket file will be re-created automatically after it is gone.
(2) If the extension is used/loaded, it will crash after the extensions_socket file disappeared. In 5.1.0, the extension will restart successfully once the extensions_socket file is re-created, but in 5.2.3/5.2.2, it will fail to restart since the extensions_socket file is still not present, and it will try to restart indefinitely.
(3) Not sure if this issue is related to #7586.

Thanks & Regards.

**Comment by Smjert** [comment]
I can definitely reproduce the issue; this is a regression for having removed a patch on Thrift which totally ignores the EINTR return when polling the socket.

The patch was designed to support the fact that when doing a privilege drop with `setuid` and `setgid`, POSIX requests that all threads have to have the same credentials.
To ensure so, the libc wrappers of those functions send a Real-Time signal (SIGRT_1) to all threads, which is caught also by the Thrift thread which is sleeping in the poll, here:
https://github.com/apache/thrift/blob/8317ec43ea2425b6f8e24e4dc4f5b2360f717eb4/lib/cpp/src/thrift/transport/TServerSocket.cpp#L590-L603
and each thread is forced to change its own credentials.

As you can see the EINTR though doesn't immediately cause the Thrift socket poll loop to exit, but it continues up to 5 times (unless the poll wakes up to read something and so the return value is success).

Now while osquery was originally using those wrappers, we moved to call the syscalls directly in this PR: https://github.com/osquery/osquery/pull/2806
Doing so avoids causing the signal to be sent; but we left some calls behind, `setgroups` and `setegid`:
https://github.com/osquery/osquery/blob/6480fc27ba55d0d4b85e61c91e1190f8add97ccf/osquery/core/system.cpp#L519
https://github.com/osquery/osquery/blob/6480fc27ba55d0d4b85e61c91e1190f8add97ccf/osquery/core/system.cpp#L535
https://github.com/osquery/osquery/blob/6480fc27ba55d0d4b85e61c91e1190f8add97ccf/osquery/core/system.cpp#L522

Changing those so that we instead call the syscall directly, seems to fix the issue without the need to patch Thrift.

This though has still to be tested/verified on macOS (when the interface to do that would change, or possibly the issue doesn't even exists, and so calling the other wrappers is ok).

**Comment by Smjert** [comment]
> @Smjert is it safe to assume that querying all these tables that use the privilege drop could cause this issue? rpm_packages (linux), deb_packages (linux), apt_sources (linux), browser_plugins (darwin), wifi(darwin)

Yes for the tables on Linux, I still haven't tested on macOS though.

**Comment by aleksmaus** [comment]
> Yes for the tables on Linux, I still haven't tested on macOS though.

I tested browser_plugins, wifi_networks tables on MacOS, can't reproduce the same issue.
